### DISCUSSION TITLE
Client Side browser default 'required' addition

### DISCUSSION THREAD

**Discussion by CoopsCodes** [goveo/react-international-phone#102] [Q&A] [answered]
Hi team,

I have been in the process of adding this to my forms as it will standardise the phone number inputs across the application.

Problem is in a few of my forms we rely on the browser default required on our main form to help progress users through inputs they missed, and while you have validation it skips by the browser default and creates a different behaviour.

What is possibility of passing required as an option for those that want to opt in?

At the moment I am stuck between having to rip out the work I have done, or remove the browser default and implement a totally custom solution.

Kind regards

Cooper

PS: this is a fantastic tool

[MDN for Client Form Validation](https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation)

**Comment by ybrusentsov** [comment]
@CoopsCodes Hey!
Have you tried to add required param to the `inputProps` prop?
I think this should solve your problem

```tsx
<PhoneInput
  defaultCountry="us"
  value={phone}
  onChange={setPhone}
  inputProps={{
    required: true,
  }}
/>
```

**Reply by CoopsCodes** [reply]
Thank you so much for getting back to me,

That unfortunately hasnt worked, I will keep trying this morning in case I have missed something, I have included my code in case you wanted to see, I am using it with Styled Components, but that shouldnt impact the functionality

```
import { Fragment } from "react";
import styled from "styled-components/macro";
import { PhoneInput } from "react-international-phone";
import { theme } from "theme";
import { Semantic } from "components/SemanticComponents";
import "react-international-phone/style.css";

const CustomPhoneInput = styled(PhoneInput)`
 ...styles omitted
`;

const Label = styled.label`
...styles omitted
`;

export default function FormInputPhoneNumber({
  defaultValue,
  onChange,
  isPhoneNumberValid,
}) {
  return (
    <Fragment>
      <Label htmlFor="mobile">
        Contact number
        <Semantic.Span color={`${theme.colors.errorRed}`}>
          {" "}
          (required)
        </Semantic.Span>
        <CustomPhoneInput
          name="mobile"
          id="mobile"
          defaultCountry="au"
          value={defaultValue}
          onChange={onChange}
          // autoComplete="tel"
          inputProps={{
            required: true,
          }}
        />
      </Label>
      {!isPhoneNumberValid && (
        <div style={{ color: "red" }}>Phone is not valid</div>
      )}
    </Fragment>
  );
}
```

**Reply by ybrusentsov** [reply]
Seems like you should put `id` and `name` to the `inputProps` too:
```jsx
<CustomPhoneInput
  defaultCountry="au"
  value={defaultValue}
  onChange={onChange}
  // autoComplete="tel"
  inputProps={{
    required: true,
    name="mobile"
    id="mobile"
  }}
/>

```

**Reply by CoopsCodes** [reply]
I added those in to the inputProps as such
```
          inputProps={{
            required: true,
            name: "mobile",
            id: "mobile",
          }}
```

But i get an error in the console:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'forEach')
    at setup (backendManager.js:4073:17)
    at welcome (backendManager.js:4057:3)
 ```
 
It might not be related to your API, but it is new when I pass more props

**Reply by CoopsCodes** [reply, marked as answer]
apologies, I have just seen it will work if the input is totally empty as such
<img width="524" alt="Screenshot 2023-09-15 at 9 34 16 am" src="https://github.com/goveo/react-international-phone/assets/46099063/1767adc6-2313-4a64-ac98-c0f11e17bd39">

But it will not recognise it if the area code is in the input.
<img width="570" alt="Screenshot 2023-09-15 at 9 33 13 am" src="https://github.com/goveo/react-international-phone/assets/46099063/da5e1a92-b2bb-4673-8714-b6e95569dec9">

Now I know this I can try with with it, this is an interesting behaviour but I guess from the browser's point of view, it isnt technically empty, I might need to put some minimum browser validation like `min="8"` etc just to grab this.

Again, thank you for your help.
